ee/models/event_definition.py

```diff
@@ -10,3 +12,26 @@ class EnterpriseEventDefinition(EventDefinition):
     tags: ArrayField = ArrayField(models.CharField(max_length=32), null=True, blank=True, default=list)
     updated_at: models.DateTimeField = models.DateTimeField(auto_now=True)
     updated_by = models.ForeignKey("posthog.User", null=True, on_delete=models.SET_NULL, blank=True)
+    verified: models.BooleanField = models.BooleanField(default=False, null=False, blank=True)
+    verified_at: models.DateTimeField = models.DateTimeField(null=True, blank=True)
+    verified_by = models.ForeignKey(
+        "posthog.User", null=True, on_delete=models.SET_NULL, blank=True, related_name="verifying_user"
+    )
+
+    def __init__(self, *args, **kwargs):
+        # Update verified_at if verified at initialization
+        super().__init__(*args, **kwargs)
+        if self.verified:
+            self.verified_at = timezone.now()
+        self._prev_verified = self.verified
+
+    def save(self, *args, **kwargs):
+        # Previously unverified event becomes verified
+        if not self._prev_verified and self.verified:
```

Comment:
```suggestion
        if not EnterpriseEventDefinition.objects.get(pk=self.pk).verified and self.verified:
```

Reply:
I'm curious, what is the advantage of using the ORM over caching the previous value? It seems cache would be slightly faster here.

Reply:
Regardless, it makes sense to do it your way if this logic is bubbled up into the serializer.

Reply:
Well I'm not sure the cache would be faster, because the ORM is lazy not sure exactly when will it fetch that value, but I think regardless it'll be the same total number of queries for the endpoint.